app/api/v_sphere/virtual_machine.rb

```diff
@@ -58,9 +58,10 @@ def self.find_by_name(name)
     end
 
     def self.prepare_vm_names
-      return [] unless File.exist?(File.join(Puppetscript.puppet_script_path, 'Node'))
+      nodes_path = Puppetscript.nodes_path
```

Comment:
How is the behavior defined if the folder not exist? In this case, the system should create the corresponding folder.     

Also the user should specify the path to the folder, in which the node/class files are stored (not to the folder where the node/classes folder is).

Reply:
What do you mean by saying
```
Also the user should specify the path to the folder, in which the node/class files are stored (not to the folder where the node/classes folder is)
```
Do you mean, when the user enters "/" as the path for the node folder, that it is automatically converted to "/Node"?
Same for the classes folder.
You are also now talking about the classes folder. This is currently the "/Name" folder. Should this be renamed to "/Class" or something similar?

Reply:
My error, everything is fine with the paths handling.